File: getCourse.go
package main

import (
	"HDU-KillCourse/client"
	"HDU-KillCourse/config"
	"HDU-KillCourse/log"
	"errors"
	"fmt"
	"strconv"
	"time"
)

func GetCourse(c *client.Client, cfg *config.Config) (*config.Course, error) {
	// 先从本地course.json读取课程信息
	courses, err := config.ReadCourse()
	if err == nil {
		return courses, nil
	}

	// 本地课程信息读取失败，从服务器获取课程信息
	log.Error("本地课程信息读取失败，正在从服务器获取课程信息...")
	// 初始化请求
	XueNian := cfg.Time.XueNian
	intXueNian, err := strconv.Atoi(XueNian)
	if err != nil {
		return nil, errors.New("学年格式错误")
	}
	xnmc := fmt.Sprintf("%s-%s", XueNian, strconv.Itoa(intXueNian+1))
	xqmc := cfg.Time.XueQi
	var xqm string
	if xqmc == "1" {
		xqm = "3"
	} else if xqmc == "2" {
		xqm = "12"
	} else {
		return nil, errors.New("学期格式错误")
	}
	req := &client.GetCourseReq{
		Cxfs:        "1",
		Zymc:        "全部",
		Xnmc:        xnmc,
		Xqmc:        xqmc,
		Kkxymc:      "全部",
		Jgmc:        "全部",
		Ywtk:        "0",
		Skfs:        "0",
		Xnm:         XueNian,
		Xqm:         xqm,
		Search:      "false",
		Nd:          fmt.Sprintf("%d", time.Now().Unix()),
		ShowCount:   "9999",
		CurrentPage: "1",
		SortOrder:   "asc",
		Time:        "0",
	}

	// 获取课程信息
	courseResp, err := c.GetCourse(req)
	if err != nil {
		return nil, err
	}
	// 转换为config.Course
	courses = &config.Course{
		Items: courseResp.Items,
	}

	// 保存课程信息到本地
	err = config.SaveCourse(courses)
	if err != nil {
		return nil, err
	}

	return courses, nil
}

File: HDU-KillCourse_Python/get_course.py
import requests
import time
import json
from HDU_Login import newjw_login

with open('config.json', 'r') as f:
    config = json.load(f)

url="https://newjw.hdu.edu.cn/jwglxt/rwlscx/rwlscx_cxRwlsIndex.html?doType=query&gnmkdm=N1548"

nd=int(time.time())
XueNian=config["time"]["XueNian"]
xnmc="{}-{}".format(XueNian,str(int(XueNian)+1))
xqmc=config["time"]["XueQi"]
if xqmc=='1':
    xqm="3"
else:
    xqm="12"

data={
    "kkbm": "",
    "kch": "",
    "kcfzr": "",
    "xsxy": "",
    "zyh_id": "",
    "bh_id": "",
    "zyfx_id": "",
    "njdm_id": "",
    "xsdm": "",
    "jxdd": "",
    "kklxdm": "",
    "xqh_id": "",
    "xkbj": "",
    "kkzt": "",
    "kclbdm": "",
    "kcgsdm": "",
    "kcxzdm": "",
    "apksfsdm": "",
    "ksfsdm": "",
    "khfsdm": "",
    "cxfs": "1",
    "jsssbm": "",
    "zcm": "",
    "xbdm": "",
    "cdlb_id": "",
    "cdejlb_id": "",
    "jxbmc": "",
    "sfzjxb": "",
    "sfhbbj": "",
    "zymc": "全部",
    "xnmc": xnmc,
    "xqmc": xqmc,
    "kkxymc": "全部",
    "jgmc": "全部",
    "njmc": "",
    "sfpk": "",
    "sfwp": "",
    "ywtk": "0",
    "skfs": "0",
    "dylx": "",
    "jzglbm": "",
    "jxms": "",
    "skpt": "",
    "sfhxkc": "",
    "sfxwkc": "",
    "sknr": "",
    "bz": "",
    "xkbz": "",
    "sfzj": "",
    "qsz": "",
    "zykfkcbj_cx": "",
    "sfgssxbk_cx": "",
    "zzz": "",
    "xf": "",
    "jys_id": "",
    "xnm": XueNian,
    "xqm": xqm,
    "js": "",
    "kclxdm": "",
    "_search": "false",
    "nd": nd,
    "queryModel.showCount": "9999",
    "queryModel.currentPage": "1",
    "queryModel.sortName": "",
    "queryModel.sortOrder": "asc",
    "time": "0"
}

session=requests.Session()
session=newjw_login.login(session)
if session is None:
    print("登录失败！")
    exit(1)

print("正在获取课程信息...")
response=session.post(url,data=data)
print("获取成功！")
response=response.json()
items=response["items"]
filtered_items = []
for item in items:
    filtered_item = {
        "jxbmc": item["jxbmc"],
        "kch_id": item["kch_id"],
        "jxb_id": item["jxb_id"],
        "jxbzc": item["jxbzc"],
        "kklxmc": item["kklxmc"],
    }
    filtered_items.append(filtered_item)
with open("hdu_kc.json","w") as f:
    json.dump(filtered_items,f,ensure_ascii=False,indent=2)
print("数据已保存到 hdu_kc.json")